src/containers/ModelPanel/selectors.ts

```diff
@@ -32,35 +36,37 @@ export const convertUITreeDataToSelectNone = createSelector(
     }
 );
 
-export const getCheckboxAllIsIntermediate = createSelector(
+// Determines if the shared checkbox should be partially checked (in "indeterminate" state)
+export const getIsSharedCheckboxIndeterminate = createSelector(
     [getUiDisplayDataTree, getAgentVisibilityMap],
-    (treeData, visibleAgents) => {
-        let childrenIntermediate = false;
+    (
+        allAgents: AgentDisplayNode[],
+        agentVisibilityMap: VisibilitySelectionMap
+    ): boolean => {
+        if (isEmpty(agentVisibilityMap)) return false;
+
+        let isAgentPartiallyVisible = false;
+        let numInvisibleAgents = 0;
+
+        // Loop through each agent and count how many are invisible. If an agent is partially
+        // visible, break out of loop and just return true
+        for (let i = 0; i < allAgents.length; i++) {
+            const agent = allAgents[i];
+            const visibleStates = agentVisibilityMap[agent.key];
 
-        // iterate through, check items with children, and also get list of
-        // agents with no children
-        const agentsWithNoChildren = treeData.filter((agent) => {
-            if (visibleAgents[agent.key] && agent.children.length) {
-                childrenIntermediate =
-                    visibleAgents[agent.key].length < agent.children.length &&
-                    visibleAgents[agent.key].length > 0;
-                return false;
-            } else if (visibleAgents[agent.key]) {
-                return true;
+            if (!visibleStates.length) {
+                numInvisibleAgents++;
+            } else if (visibleStates.length < agent.children.length) {
+                isAgentPartiallyVisible = true;
+                break;
```

Comment:
cant you just return true here, and then you don't need the `isAgentPartiallyVisible` at all?

Reply:
Yes totally! Thanks! Fixed